main.cpp

```diff
@@ -270,13 +178,102 @@ struct FloatType
         value = nullptr;
     }
     
+    FloatType& pow(const IntType& x);
+    FloatType& pow(const FloatType& x);
+    FloatType& pow(const DoubleType& x);
+    FloatType& pow(float v);
+    
+    FloatType& powInternal(float x);
```

Comment:
This should be `private`.  if it's not private, it's not internal!
- [ ] revise